src/test/java/sysc4806/graduateAdmissions/controller/ApplicationControllerTest.java

```diff
@@ -73,16 +77,18 @@ void setUpMocks() {
         resumeFileName = "resume.pdf";
 
         applications = Arrays.asList(
-                new Application(0, applicant1, term, Department.ELEC, degree,  professors, status, gpa, resumeFileName),
-                new Application(1, applicant1, term, Department.SYSC, degree,  professors, status, gpa, resumeFileName),
-                new Application(2, applicant2, term, Department.SREE, degree,  professors, status, gpa, resumeFileName));
-
-        when(repo.findAll()).thenReturn(applications);
-        when(repo.findByApplicant_id(applicant1.getId())).thenReturn(applications.subList(0, 2));
-        when(repo.findByApplicant_id(applicant2.getId())).thenReturn(Collections.singletonList(applications.get(2)));
+                new Application(0, term, Department.ELEC, degree,  professors, status, gpa, resumeFileName),
+                new Application(1, term, Department.SYSC, degree,  professors, status, gpa, resumeFileName),
+                new Application(2, term, Department.SREE, degree,  professors, status, gpa, resumeFileName));
+        applicant1.setApplications(new HashSet<>(applications.subList(0, 2)));
+        applicant2.setApplications(new HashSet<>(Collections.singletonList(applications.get(2))));
+
+        when(applicationRepository.findAll()).thenReturn(applications);
+        when(userRepository.findById(applicant1.getId())).thenReturn(Optional.ofNullable(applicant1));
+        when(userRepository.findById(applicant2.getId())).thenReturn(Optional.ofNullable(applicant2));
```

Comment:
This invalidates most of the testing done here, but we're short on time and I'd like to start working again.

No changes required for this PR